woocommerce/woocommerce-android
# Hack week: Update target SDK from 31 to 33

This PR updates the target SDK from 31 to 33 and makes the [necessary changes](https://developer.android.com/about/versions/13/behavior-changes-13):

- [Moved the package namespace declarations from the manifests to gradle files](https://github.com/woocommerce/woocommerce-android/commit/e95470306749361bac7474d544ad312226216321)
- ~~[Updated Gradle version to 7.3.0, which addressed the warning caused by the target SDK update](https://github.com/woocommerce/woocommerce-android/commit/2ee0173cf78bae8fc867b513fead78a78ab336de)~~
- Implemented a notification permission request bar, which is displayed now if the app does not have the permission to send push notifications. The users can dismiss the bar and they won't be bothered by it afterwards. This changes the current behavior, which displayed the permission request on devices running Android 13 (API 33) immediately after the app launch. [More info here](https://developer.android.com/develop/ui/views/notifications/notification-permission).

| Request bar | Permission dialog |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/1522856/224506452-46508227-9fa1-414e-9b63-d2609c9bf3c7.png) | ![image](https://user-images.githubusercontent.com/1522856/224506482-9a53dd13-dbf2-4098-8c4f-fd69c6dcebdd.png) |


[More changes are required](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions) in the Media library, which are addressed in a [separate PR](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/63).

**To test:**

You'll need a device running Android version 13 (API level 33) to test the new bahavior.

1. Start the app with a clean installation
2. Notice the notifications permission request is not displayed immediately after the launch anymore
3. Proceed with the login
4. Notice that a banner pops up at the bottom with a description of the permission request
5. Tap on Allow
6. Grant the notifications permission
7. Kill the app and start it again
8. Notice the banner doesn't pop up
9. Go to App info -> Permissions
10. Verify that `Notifications` is listed in the `Allowed` section
11. Go to the Storage settings and clear the app storage
12. Go back to the app and log in
13. Notice the permission banner pops up again
14. Tap on Dismiss
15. Kill the app and start it again
16. Notice the app doesn't pop up
17. Go to App info -> Permissions
18. Verify that `Notifications` is listed in the `Not Allowed` section
